pkg/client/metadata/metadata.go

```diff
@@ -43,9 +47,13 @@ func (u classicEnvURL) GetURL() string {
 	return u.Domain
 }
 
-// GetDynatraceClassicURL tries to fetch the URL of the classic environment using the API of a platform enabled
-// environment
+// GetDynatraceClassicURL tries to fetch the URL of the classic environment using the API of a platform enabled environment.
+// The call has a timeout of 15 seconds.
 func GetDynatraceClassicURL(ctx context.Context, platformClient corerest.Client) (string, error) {
+	ctx = logr.NewContext(ctx, log.WithCtxFields(ctx).GetLogr())
+	ctx, cancel := context.WithTimeout(ctx, 15*time.Second)
+	defer cancel()
```

Comment:
Do we even need cancel here? The context is only used for the API call and after that it's not used anymore, so `cancel` would noch make any difference

Reply:
I think so: the pattern from the docs for `context.WithTimeout(...)`:
```
func slowOperationWithTimeout(ctx context.Context) (Result, error) {
  ctx, cancel := context.WithTimeout(ctx, 100*time.Millisecond)
  defer cancel()  // releases resources if slowOperation completes before timeout elapses
  return slowOperation(ctx)
}
```